src/libs/actions/IOU.js

```diff
@@ -660,28 +765,37 @@ function deleteMoneyRequest(chatReportID, iouReportID, moneyRequestAction, shoul
         iouReportID,
     );
 
-    const currentUserEmail = optimisticReportAction.actorEmail;
-    const updatedIOUReport = IOUUtils.updateIOUOwnerAndTotal(iouReport, currentUserEmail, amount, moneyRequestAction.originalMessage.currency, CONST.IOU.REPORT_ACTION_TYPE.DELETE);
-    chatReport.lastMessageText = optimisticReportAction.message[0].text;
-    chatReport.lastMessageHtml = optimisticReportAction.message[0].html;
-    chatReport.hasOutstandingIOU = updatedIOUReport.total !== 0;
+    const currentUserEmail = optimisticIOUAction.actorEmail;
+    let updatedIOUReport = {};
+    if (ReportUtils.isExpenseReport(iouReportID)) {
+        updatedIOUReport = {...iouReport};
+
+        // Because of the Expense reports are stored as negative values, we add the total from the amount
+        updatedIOUReport.total += amount;
+    } else {
+        updatedIOUReport = IOUUtils.updateIOUOwnerAndTotal(
+            iouReports[`${ONYXKEYS.COLLECTION.REPORT}${iouReportID}`],
```

Comment:
You got the iouReport defined above
```suggestion
            iouReport,
```